### ISSUE TITLE
Tslint ignores typeRoots in tsconfig.json.

### ISSUE THREAD

**Issue by resurge** [palantir/tslint#3962]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: CLI (via create-react-app-typescript's start script, I assume this is CLI)

#### TypeScript code being linted
I made a repo here: https://github.com/resurge/tslint-ignores-tsconfig-types

index.tsx
```ts
window.myGlobalVariable = "global variable value";
```

types/global.d.ts
```ts
interface Window {
	myGlobalVariable: string;
}
```

tsconfig.json
```ts
{
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "build/dist",
    "module": "esnext",
    "target": "es5",
    "lib": ["es6", "dom"],
    "sourceMap": true,
    "allowJs": true,
    "jsx": "react",
    "moduleResolution": "node",
    "rootDir": "src",
    "forceConsistentCasingInFileNames": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "suppressImplicitAnyIndexErrors": true,
    "noUnusedLocals": true,
    "typeRoots": [
      "./node_modules/@types",
      "./types"
    ]
  },
  "exclude": [
    "node_modules",
    "build",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "src/setupTests.ts"
  ]
}

```

with `tslint.json` configuration:
(Standard create-react-app-typescript configuration)
```json
{
  "extends": ["tslint:recommended", "tslint-react", "tslint-config-prettier"],
  "linterOptions": {
    "exclude": [
      "config/**/*.js",
      "node_modules/**/*.ts"
    ]
  }
}
```

#### Actual behavior
Tslint gives the following error:
> Property 'myGlobalVariable' does not exist on type 'Window'.

Even though the type definitions are defined and referfenced in tsconfig.json

#### Expected behavior
Tslint doesn't give any errors.

#### Workaround
If I use the following code in index.tsx and remove the typeRoots config it works.
```ts
declare global {
  // tslint:disable-next-line
  interface Window {
    myGlobalVariable: string;
  }
}
```
(The disable-next-line is because create-react-app-typsecript requires interfaces to start with I)

**Comment by joonhocho** [comment]
I am having the same problem. typeRoots in tsconfig.json is not honored by tslint.

**Comment by JoshuaKGoldberg** [comment]
Perhaps also related: #3931

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
